lineapy/graph_reader/program_slice.py

```diff
@@ -49,11 +60,65 @@ def get_slice_graph(
     for sink in sinks:
         ancestors.update(graph.get_ancestors(sink))
 
+    ancestors = fill_dangling_nodes_in_graph(graph, ancestors)
+
     new_nodes = [graph.ids[node] for node in ancestors]
     subgraph = graph.get_subgraph(new_nodes)
     return subgraph
 
 
+def fill_dangling_nodes_in_graph(
```

Comment:
in general have a lot more documentation on what this does and how it achieves it.

Reply:
renaming function suggested. make it descriptive if possible. also have docstring explain what it does before params.